### ISSUE TITLE
Legacy Database

### ISSUE THREAD

**Issue by A-L33** [GSA/smartpay-training#113]
The GSA SmartPay program [decided](https://github.com/GSA/smartpay-training/issues/113#issuecomment-1516546746) to _not_ migrate its legacy database. However, the program will still need the ability to access the legacy data for some period of time (up to 7 years).

Dependency on legacy site retirement and final data download from legacy contractor host.

The features in particular that will be needed are:

- [ ] Ability to pull training reports
- [ ] Ability to pull a copy of a certificate (to be emailed to user) (nice to have)

**Comment by A-L33** [comment]
@JessicaMarine1 As we discussed yesterday.  This is just a starting point for now, we can look at this more as we get closer to the required task.

**Comment by JessicaMarine1** [comment]
4.20.2023 - Decision was made to prioritize adding bureaus to the system over migrating legacy data. SmartPay will likely need the SD's team to help them figure out a way to access the legacy data so that they can respond to requests from users (for old certificates) and A/OPCs (for reporting), but no full migration planned for MVP. Migrating legacy data will be planned for post-launch, subject to availability of funding and need.

**Comment by JessicaMarine1** [comment]
Mark: When we first got started, we got a data dump from vendor and were able to get a local version running on our machines. _Should_ be able to get a new copy and stand it up and put it behind a little login.

Andrew: What security requirements would this need?

Jeff: I think we should take a closer look at the format of the legacy data and see if we can extract it into a generic format that we can query with some simple tools rather than preserving Drupal

**Comment by rebekahperillo** [comment]
Mark, Tri, Jessica, Andrew, and I had a discussion today about legacy data. Arpan joined midway through the meeting.

For the program website, it appears as if we will download the HTML from the current website and upload a directory of its information to GitHub (links, documents, etc). The program website is public, so there is no issue housing its data on GitHub (which is available to the public).

For the training website, we will ask for a download from the current contractor. We are hoping to receive the go-ahead to store it on a Google drive, encrypt it, and restrict access to the CCCM team only. Rebekah is going to reach out to the Google POC at GSA to double check that the file size will work on Google drive.

If I forgot anything, @A-L33, feel free to jump in. :)

**Comment by rebekahperillo** [comment]
**Rebekah's email to Chris McFerran - Director, Enterprise Systems Support Division (ICE), Office of Corporate IT Services, GSA IT:**

"In preparation for launch, we're trying to figure out how to store legacy data from our current training site (names and emails for completed training classes). We estimate that the file size is about 700 MB.

We are interested in possibly storing the file on a Google drive, encrypting it, and restricting access to fellow team members only. Do you see any issue with this possible path forward? Does the file size cause any issue?"

**Chris's response about storing the data, encrypting the data, and restricting access to the data:**

"Here is a collection of comments from a couple people on my team:

Not sure the reason for encryption but the access can be restricted to the drive to prevent unauthorized access. But if they really want to encrypt the files, I would recommend zipping the file and encrypting it before uploading it to Google drive. We have no other way to encrypt the data at rest.

Agreed, the data itself is encrypted on Google server drives, but if they want it protected against someone accidentally getting access and seeing the information, zip with encryption would be the way to go.

Google encrypts data at rest so it's not necessary to encrypt the files again and the ACL of the drive folder will protect the file from unauthorized access. But if the customer wants to encrypt the files and then upload to Google Drive, they can do that for their own satisfaction.

Customer should note, if they decide to encrypt the files before uploading to Google drive then they need to keep the decryption key somewhere safe as they will be responsible for that. We are unable to help if they lose decryption key."

**Chris's response about the data's file size:**
"Drive upload & copy limits
Each user can upload and copy 750 GB to Drive within 24 hours. Users who reach the 750-GB limit can't upload or copy more files until the limit refreshes. The limit refreshes within 24 hours.

You can upload and synchronize files up to 5 TB in size. Only the first file that breaks the limit completes uploading.
Files larger than 750 GB can’t be copied. Instead, download the file and then upload it.

The source of the above information is https://support.google.com/a/answer/172541?hl=en&sjid=14297886250930114448-NA"

**Comment by rebekahperillo** [comment]
Also, for the training site, we should also consider getting copies of the course material that resided on the site (in addition to obtaining the legacy data).

**Comment by JessicaMarine1** [comment]
@A-L33 @rebekahperillo moving to QA to have the program test it out over the next week or so to make sure it's meeting you needs:

- Database provided by the vendor seems to be sufficient.
- Program to confirm:
  - Can search / find user's information
  - Can filter to provide agency reports
  - Can produce a certificate, if needed (legacy vendor to provide template)

**Comment by rebekahperillo** [comment]
@A-L33 @JessicaMarine1 I'm just waiting on the last AC criteria. I'd like to see the templates, before I approve. Thank you!

**Comment by rebekahperillo** [comment]
Mark provided copies/templates for the 6 types of certificates. Moving to DONE. Thank yoU!
